File: argentum-ui/src/app/common/rest-service/test-data/discounts.ts
import {Discount} from '../../model/discount';
import {Statuses} from './statuses';
import {Categories} from './categories';

export namespace Discounts {
  export const PAID_SOFT_DRINKS = new Discount(22010, Statuses.PAID, Categories.SOFT_DRINKS, 0.10);
  export const PENDING_HARD_DRINKS = new Discount(22020, Statuses.PENDING, Categories.HARD_DRINKS, 0.25);

  export const ALL = [PAID_SOFT_DRINKS, PENDING_HARD_DRINKS];

  export const PENDING_SOFT_DRINKS = new Discount(22030, Statuses.PENDING, Categories.SOFT_DRINKS, 0.20);
  export const PAID_SOFT_DRINKS_PATCHED_REQUEST = new Discount(22010, undefined, undefined, 0.40);
  export const PAID_SOFT_DRINKS_PATCHED_RESPONSE = new Discount(22011, Statuses.PAID, Categories.SOFT_DRINKS, 0.40);
}

File: argentum-api/api/tests/data/discounts.py
from decimal import Decimal

from api.models.discount import Discount
from api.tests.data.categories import TestCategories
from api.tests.data.statuses import TestStatuses
from api.tests.utils.test_objects import TestObjects


class TestDiscounts(TestObjects):
    MODEL = Discount

    PAID_SOFT_DRINKS: MODEL
    PENDING_HARD_DRINKS: MODEL

    PENDING_SOFT_DRINKS: MODEL
    PAID_SOFT_DRINKS_PATCHED: MODEL

    @classmethod
    def init(cls):
        cls.PAID_SOFT_DRINKS = cls.MODEL(
            id=22010,
            status=TestStatuses.PAID,
            category=TestCategories.SOFT_DRINKS,
            rate=Decimal('0.10')
        )

        cls.PENDING_HARD_DRINKS = cls.MODEL(
            id=22020,
            status=TestStatuses.PENDING,
            category=TestCategories.HARD_DRINKS,
            rate=Decimal('0.25')
        )

        cls.SAVED = [cls.PAID_SOFT_DRINKS, cls.PENDING_HARD_DRINKS]

        cls.PENDING_SOFT_DRINKS = cls.MODEL(
            id=22030,
            status=TestStatuses.PENDING,
            category=TestCategories.SOFT_DRINKS,
            rate=Decimal('0.20')
        )

        cls.PAID_SOFT_DRINKS_PATCHED = cls.MODEL(
            id=22011,
            status=TestStatuses.PAID,
            category=TestCategories.SOFT_DRINKS,
            rate=Decimal('0.40')
        )

        cls.UNSAVED = [cls.PENDING_SOFT_DRINKS, cls.PAID_SOFT_DRINKS_PATCHED]
